Newspack/NewspackFramework/Shadows/ShadowManager.swift

```diff
@@ -0,0 +1,115 @@
+import Foundation
+
+/// Manages saving and retrieving shadow sites and assets from shared defaults
+/// and wrangling files in shared storage.
+///
+public class ShadowManager {
+
+    public init() {}
+
+    /// Stores the passed array of shadow sites in shared defaults.
+    /// This will overwrite whatever information is currently stored.
+    ///
+    /// - Parameter sites: An array of ShadowSite objects.
+    ///
+    public func storeShadowSites(sites: [ShadowSite]) {
+        // create empty shadow array
+        // Get sites
+        // for each site
+        // get folders
+        // for each folder
+        // generate shadow folder
+        // generate shadow site
+        // get shadow site dictionary
+
+        // add shadow site dictionary to shadow array
+        var arr = [[String: Any]]()
+
+        for site in sites {
+            arr.append(site.dictionary)
+        }
+
+        // save shadow array in user defaults
+        UserDefaults.shared.set(arr, forKey: AppConstants.shadowSitesKey)
```

Comment:
(Just in case!!) is `site.dictionary` always gonna be returning raw types?

Reply:
Yep. It should always return a format that can be serialized to defaults.